engine/src/main/java/org/apache/apex/engine/util/PubSubWSClientBuilder.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.engine.util;
+
+import java.net.URISyntaxException;
+
+import com.datatorrent.api.Context;
+import com.datatorrent.stram.plan.logical.LogicalPlan;
+import com.datatorrent.stram.util.PubSubWebSocketClient;
+import com.datatorrent.stram.util.SharedPubSubWebSocketClient;
+
+public class PubSubWSClientBuilder
+{
+  public static final String GATEWAY_LOGIN_URL_PATH = "/ws/v2/login";
+
+  private Context context;
+
+  public PubSubWSClientBuilder setContext(Context context)
+  {
+    this.context = context;
+    return this;
+  }
+
+  public <T extends PubSubWebSocketClient> T build(Class<T> clazz) throws URISyntaxException
+  {
+    // The builder can be used to build different types of PubSub clients in future but for now only one is supported
```

Comment:
Since there is already a hierarchy of PubSub and SharedPubSub I assumed there is an intent to have multiple PubSubClients. I can make it private for now.

Reply:
The method is capable of constructing SharedPubSubWebSocketClient only, and it is not part of API anyway, why to support different classes in method signature only? When there is a request to support building PubSubWebSocketClient, the method can be extended as needed.

Reply:
The idea is the method content will change when new PubSubWebSocketClient implementations are available. The method signature can remain the same. Once it is private, I will remove the initial check for the argument.

Reply:
OK